src/main/java/com/commercetools/sync/categories/CategorySync.java

```diff
@@ -0,0 +1,190 @@
+package com.commercetools.sync.categories;
+
+import com.commercetools.sync.categories.helpers.CategorySyncOptions;
+import com.commercetools.sync.categories.helpers.CategorySyncStatistics;
+import com.commercetools.sync.categories.utils.CategorySyncUtils;
+import com.commercetools.sync.commons.Sync;
+import io.sphere.sdk.categories.Category;
+import io.sphere.sdk.categories.CategoryDraft;
+import io.sphere.sdk.commands.UpdateAction;
+import io.sphere.sdk.models.SphereException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.List;
+
+import static java.lang.String.format;
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+
+public class CategorySync implements Sync {
+    private static final Logger LOGGER = LoggerFactory.getLogger(CategorySync.class);
+    private final CategorySyncOptions syncOptions;
+    private final CategorySyncStatistics statistics;
+
+    public CategorySync(@Nonnull final CategorySyncOptions syncOptions) {
+        this.syncOptions = syncOptions;
+        statistics = new CategorySyncStatistics();
```

Comment:
Here time measurement in your statistics becomes invalid. Why? It doesn't measure only "processing time" but sum of processing time and time that passed between `CategorySync` instance creation and calling for sync method. I can imagine situation where user creates Syncer instance, then prepares his drafts and finally calls sync method. The processing time would be completely wrong then...